codec-http2/src/main/java/io/netty/handler/codec/http2/DefaultHttp2Headers.java

```diff
@@ -175,6 +176,11 @@ protected void validateName(NameValidator<CharSequence> validator, boolean forAd
     protected void validateValue(ValueValidator<CharSequence> validator, CharSequence name, CharSequence value) {
         // This method has a noop override for backward compatibility, see https://github.com/netty/netty/pull/12975
         super.validateValue(validator, name, value);
+        // https://datatracker.ietf.org/doc/html/rfc9113#section-8.3.1
+        if ((value == null || value.length() == 0) && hasPseudoHeaderFormat(name) && getPseudoHeader(name) == PATH) {
```

Comment:
We need to protect this validation by a configuration flag to make sure users can disable it.

Taking into account the RFC requirements to always run this validation (at least by default), the fact that `validateValues == false` by default, I would propose to use `validate` flag that controls header name validation. While in practice the check validates a value, I would say that technically it's a "validation of a preudo-header name". The purpose of `validateValues` is quite different. It's disabled by default because it will cause significant perf overhead for most users.

Reply:
Please check the new commit.

Reply:
The biggest differences are:
1. the cost associated with a check
2. how likely is it to hit the problem (messing up with pseudo-headers seems more likely)
3. how it will affect the h2-protocol related flow vs user's business logic (having a bad char in some header won't impact h2 codec or h2->h1 transition)